### PR TITLE
Upload file using SDK instead of XHR

### PR DESCRIPTION
**Pull request by PVince81**
## Description
Use owncloud-sdk for uploading files

## Related Issue
None raised

## Motivation and Context
As we'll add chunk upload later to the SDK, it makes sense to use the SDK for upload instead.
It will only be a matter of adding "maxChunkSize" to the options to have it using chunks.
I don't think we should implement chunking directly in Phoenix.

Also: as we use the SDK in other places we should be consistent and use it everywhere!

## How Has This Been Tested?
Quick manual test with a file upload:
- [x] TEST: upload works in root :robot: 
- [x] TEST: upload works in subdir :robot: 
- [x] TEST: upload works in public page root :robot: 
- [x] TEST: upload works in public page subdir :robot: 
- [x] TEST: upload works in files drop :robot: 
- [x] TEST: progress bar works
- [ ] TEST: overwrite works in all above cases
- [x] TEST: upload with drag and drop in all the above cases
- [ ] TEST: IE11...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
- [x] REQUIRES: owncloud-sdk merge https://github.com/owncloud/owncloud-sdk/pull/366 for progress events
- [x] REQUIRES: davclient.js merge https://github.com/owncloud/davclient.js/pull/39 for progress events
- [x] TODO: also apply and test the change for public file upload
- [x] TODO: delete FileUpload.js

### CODE REVIEW

**Comment by PVince81** [comment]
@LukasHirt @DeepDiver1975 please have a look at this POC and also the linked dependencies in owncloud-sdk and let me know if this is ok.

I intend to delete FileUpload.js (but not FileUpload.vue) and later on we can implement chunking inside of owncloud-sdk: https://github.com/owncloud/owncloud-sdk/issues/18

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5702/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>iPhone</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5702/

<p>









</p></details>

File: apps/files/src/mixins.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import filesize from 'filesize'
 import moment from 'moment'
-import FileUpload from './FileUpload.js'
```
Comment (DeepDiver1975) [inline apps/files/src/mixins.js]: can we delete FileUpload.js?

Comment (PVince81) [reply apps/files/src/mixins.js]: yes, when I'm done. See TODO list in top post

**Comment by PVince81** [comment]
also we need to talk about "loadend" now (see conflict), I'm still not convinced that we need it.
especially we can't add it in davclient.js as that library has no polyfills in place and adding those there seems like overkill

**Comment by PVince81** [comment]
seems loadend also fires after an error: https://humanwhocodes.com/blog/2012/05/22/working-with-files-in-javascript-part-3/

**Comment by PVince81** [comment]
more trouble: uploading into subdirs doesn't work because all the upload code assumptions so far was about using the full URL as prefix. this means we need to change many places to use the current folder for its path. This changes the paradigms a bit...

- [ ] change URL use for upload to using path instead

**Comment by PVince81** [comment]
I managed to fix the subdir upload but I'm now blocked as I can't test public upload: need #1951 first

**Comment by PVince81** [comment]
note: acceptance tests to be added as part of ticket https://github.com/owncloud/phoenix/issues/2222

**Comment by LukasHirt** [comment]
> seems loadend also fires after an error: https://humanwhocodes.com/blog/2012/05/22/working-with-files-in-javascript-part-3/

Exactly, that's what I mentioned here https://github.com/owncloud/phoenix/pull/2225#discussion_r334413669 and why I used it at all.

**Comment by LukasHirt** [comment]
Anyway, like a lot the approach to move it into the sdk :+1:

**Comment by PVince81** [comment]
I've fixed the public link upload. I find some messy stuff there like the token sometimes being embedded in the path... I'll raise a separate ticket to address this debt as I'd prefer to have more transparency there where the file list components don't need to care about public or not. This might affect how SDK API is designed.

- [ ] wait for SDK update + package.json + locks update
- [ ] raise ticket for public link code mess (+ bugs with createFolder not accepting password...)

**Comment by PVince81** [comment]
- [ ] rebase once the public upload acceptance tests are in place, as they will confirm that this works correctly
- [ ] solve conflicts

**Comment by PVince81** [comment]
all done. I did a quick run of some upload tests locally and they passed.

let's see if the whole suites run...

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6469/

<p>





</p></details>

**Comment by PVince81** [comment]
- [ ] BUG: upload issues on public link page:  tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature:206 and others

**Comment by PVince81** [comment]
Fixed files drop case. Glad to see that this could be caught by acceptance tests!

**Comment by PVince81** [comment]
@LukasHirt all green in CI, just some bug that prevented it to appear published on Github.

Can we merge ?

**Comment by PVince81** [comment]
- [ ] BUG: drag and drop from file manager onto the drop zone (overlay that appears when dragging) => it currently opens the file directly ?!

**Comment by PVince81** [comment]
Fixed drag and drop bug. It turns out that vue-dropzone needs a URL even though we don't use that functionality. I've fed it a dummy URL for now. To be cleant up in https://github.com/owncloud/phoenix/issues/1953

**Comment by PVince81** [comment]
One CI job got stuck on cloning. Restart build.

**Pull request merged**